packages/powersync_attachments_stream/lib/src/abstractions/attachment_context.dart

```diff
@@ -0,0 +1,49 @@
+import '../attachment.dart';
+
+/// Context for performing Attachment operations.
+///
+/// This is typically provided through a locking/exclusivity method and allows
+/// safe, transactional operations on the attachment queue.
+abstract class AbstractAttachmentContext {
```

Comment:
I don't really understand the purpose of making this and `AbstractAttachmentService` an abstract class:

1. It doesn't look like it makes sense for users to write their own implemenation of this, so by exposing this interface we can't change or add methods without a breaking change.
2. It means that users have to manually create `AttachmentServiceImpl`, it would be nicer if the implementation class was just named `AttachmentService` directly and we didn't have a separate interface.